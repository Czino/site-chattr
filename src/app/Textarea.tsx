'use client'
import { TextareaHTMLAttributes } from 'react'

export const Textarea = (props: TextareaHTMLAttributes<HTMLTextAreaElement>) => (
    <textarea
        {...props}
        className="resize-none border border-gray-300 rounded-md p-2 focus:outline-none focus:ring-2 focus:ring-blue-500 flex-shrink-0"
    />
)
