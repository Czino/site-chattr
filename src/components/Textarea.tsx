import { ChangeEvent, TextareaHTMLAttributes } from 'react'

type Props = TextareaHTMLAttributes<HTMLTextAreaElement> & {
    onTextChange?: (value: string) => void
}
export const Textarea = ({ onTextChange, className, ...props }: Props) => {
    const handleChange = (event: ChangeEvent<HTMLTextAreaElement>) => {
        if (onTextChange) onTextChange(event.target.value)
    }
    return (
        <textarea
            {...props}
            onChange={handleChange}
            className={[
                'resize-none border border-gray-300 rounded-md p-2',
                'focus:outline-none focus:ring-2 focus:ring-blue-500',
                className,
            ].join(' ')}
        />
    )
}
