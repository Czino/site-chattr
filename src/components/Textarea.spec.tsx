import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Textarea } from './Textarea'

describe('Textarea', () => {
    const onTextChange = jest.fn()
    it('should render Textarea with styles', () => {
        const { asFragment } = render(<Textarea className="mt-4" value="Text" onTextChange={onTextChange} />)
        expect(asFragment()).toMatchSnapshot()
    })
    it('should call onTextChange handler on text change', async () => {
        render(<Textarea placeholder="Your message..." onTextChange={onTextChange} />)
        await userEvent.type(screen.getByPlaceholderText('Your message...'), 'a message')
        expect(onTextChange).toHaveBeenCalledWith('a message')
    })
    it('should do nothing if onTextChange handler not given', async () => {
        render(<Textarea placeholder="Your message..." />)
        await userEvent.type(screen.getByPlaceholderText('Your message...'), 'a message')
    })
})
