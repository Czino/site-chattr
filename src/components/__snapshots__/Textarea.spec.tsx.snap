// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea should render Textarea with styles 1`] = `
<DocumentFragment>
  <textarea
    class="resize-none border border-gray-300 rounded-md p-2 focus:outline-none focus:ring-2 focus:ring-blue-500 mt-4"
  >
    Text
  </textarea>
</DocumentFragment>
`;
